odoo/models.py

```diff
@@ -5783,9 +5786,8 @@ def process(field):
 
         if fnames is None:
             # recompute everything
-            fields_to_compute = self.env.fields_to_compute()
-            while fields_to_compute:
-                process(next(iter(fields_to_compute)))
+            for field in list(self.env.fields_to_compute()):
+                process(field)
```

Comment:
So there was no real reason for the old one to use a `while`, nothing can be *added* to the set of fields to compute as we're iterating on it?

Reply:
We use the transitive closure of dependencies for triggering recomputations.  So nothing new should be marked to recompute while processing recomputations.  That is why the `for` loop should be good enough.

Reply:
ok